app/main/views/audits.py

```diff
@@ -180,6 +212,52 @@ def acknowledge_audit(audit_id):
     return jsonify(auditEvents=audit_event.serialize()), 200
 
 
+@main.route('/audit-events/<int:audit_id>/acknowledge-including-previous', methods=['POST'])
+def acknowledge_including_previous(audit_id):
+    updater_json = validate_and_return_updater_request()
+
+    audit_event = AuditEvent.query.get(audit_id)
+    if audit_event is None:
+        abort(404, "No audit event with this id")
+
+    result = db.session.execute(AuditEvent.__table__.update().returning(
+        AuditEvent.__table__.c.id
+    ).where(db.and_(
+        AuditEvent.__table__.c.object_id == audit_event.object_id,
+        AuditEvent.__table__.c.object_type == audit_event.object_type,
+        AuditEvent.__table__.c.type == audit_event.type,
+        AuditEvent.__table__.c.acknowledged == db.false(),
+        # ugly, but this just implements the same "id tie breaker" behaviour for created_at-equal events as the
+        # ordering we use in list_audits. this way there is some consistency between the two views as to what events
+        # are considered "previous" in such cases. we could use postgres composite types to express this far more
+        # neatly, but i can't get sqlalchemy to work with anonymous composite types.
+        db.or_(
+            AuditEvent.__table__.c.created_at < audit_event.created_at,
+            db.and_(
+                AuditEvent.__table__.c.created_at == audit_event.created_at,
+                AuditEvent.__table__.c.id <= audit_event.id,
+            ),
+        ),
+    )).values(
+        acknowledged=db.true(),
+        acknowledged_at=datetime.utcnow(),
+        acknowledged_by=updater_json['updated_by'],
+    )).fetchall()
+    db.session.expire_all()
+
+    try:
+        db.session.commit()
+
+    except IntegrityError as e:
+        db.session.rollback()
+        abort(400, e.orig)
```

Comment:
Why bother catching this? There's no particular evidence that it's the client's fault is there (which would motivate the 400 rather than a 500), and if we don't catch it the rollback will happen anyway?